SocketRocket/SRWebSocket.m

```diff
@@ -373,6 +373,25 @@ - (void)_HTTPHeadersDidFinish;
 {
     NSInteger responseCode = CFHTTPMessageGetResponseStatusCode(_receivedHTTPHeaders);
     if (responseCode >= 400) {
+        
+        if (responseCode == 401)
+        {
+            // handle an aunathorised request to upgrade
+            if ([self.authenticationDelegate respondsToSelector:@selector(webSocket:bearerAuthenticationDidFailWithRetryHandler:)])
+            {
+                [self.authenticationDelegate webSocket:self bearerAuthenticationDidFailWithRetryHandler:^(NSString *newBearerToken) {
+                    if (newBearerToken)
+                    {
+                        dispatch_async(_workQueue, ^{
+                            self.authenticationBearerToken = newBearerToken;
+                            [self didConnect];
```

Comment:
Shouldn't really need to use weakSelf as the block doesn't have a copy reference to it.